### ISSUE TITLE
False positive for E1123 (unexpected-keyword-arg)

### ISSUE THREAD

**Issue by bje-** [pylint-dev/pylint#3893]
### Steps to reproduce
1. Run pylint over the following test case:

```python
"""docstring"""

# pylint: disable=too-few-public-methods

class PV1Axis():
    """Single-axis tracking PV."""
    def __init__(self, polygon, capacity, filename, column, build_limit=None, label='PV'):
        pass

class Wind():
    """Wind farm."""
    def __init__(self, polygon, capacity, filename, column, delimiter=None,
                 build_limit=None, label='wind'):
        pass

for g in [PV1Axis, Wind]:
    if g == Wind:
        g(0, 0, 'foo', 0, delimiter=',', build_limit=100, label='wind')
```

### Current behavior

```
example.py:18: [E1123(unexpected-keyword-arg), ] Unexpected keyword argument 'delimiter' in constructor call
```

### Expected behavior

I don't expect to see any warnings here.

### pylint --version output

```
pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0]
```

**Comment by hippo91** [comment]
@bje- thanks for the report. I can reproduce it. 
I think this issue is linked to the fact that `pylint` doesn't have control flow capabilities. 
And interesting thing is that the following code doesn't trigger any warning:
```
"""docstring"""

# pylint: disable=too-few-public-methods, too-many-arguments

class PV1Axis():
    """Single-axis tracking PV."""
    def __init__(self, polygon, capacity, filename, column, build_limit=None, label='PV'):
        pass

class Wind():
    """Wind farm."""
    def __init__(self, polygon, capacity, filename, column, delimiter=None,
                 build_limit=None, label='wind'):
        pass

for g in [Wind, PV1Axis]:
    if g == Wind:
        g(0, 0, 'foo', 0, delimiter=',', build_limit=100, label='wind')
        Wind(0, 0, 'foo', 0, delimiter=',', build_limit=100, label='wind')
```

**Comment by bje-** [comment]
This was fixed somewhere along the way.

**Comment by Pierre-Sassoulas** [comment]
Are you sure you did not deactivate ``unexpected-keyword-arg`` @bje- ? I can reproduce on latest main with ``pylint a.py --enable=all --enable-all-extensions``

**Comment by bje-** [comment]
It works for me!

```
(myenv) bje:~/source/nemo (master)$ pylint --version
pylint 2.14.5
astroid 2.12.2
Python 3.10.6 (main, Aug  2 2022, 15:11:28) [GCC 9.4.0]
(myenv) bje:~/source/nemo (master)$ pylint test.py --enable=all --enable-all-extensions

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

**Comment by Pierre-Sassoulas** [comment]
Hmm pylint 2.14.5 is not supposed to work with astroid 2.12.2 ("pylint 2.14.5 requires astroid<=2.12.0-dev0,>=2.11.6, but you have astroid 2.12.2 which is incompatible."), and even when "forcing" the install I still have ``E1123: Unexpected keyword argument 'delimiter' in constructor call (unexpected-keyword-arg)``, that's very strange.

**Comment by bje-** [comment]
Ubuntu 22.04
